src/main/java/eu/miaplatform/service/Service.java

```diff
@@ -0,0 +1,114 @@
+package eu.miaplatform.service;
+
+import okhttp3.*;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+public class Service {
+    private static final String APPLICATION_JSON_CHARSET_UTF_8 = "application/json; charset=utf-8";
+    private static final MediaType JSON = MediaType.parse(APPLICATION_JSON_CHARSET_UTF_8);
+    private String serviceName;
+    private Headers headers;
+    private InitServiceOptions options;
+
+    private OkHttpClient client = new OkHttpClient.Builder()
+            .connectionSpecs(Arrays.asList(ConnectionSpec.MODERN_TLS, ConnectionSpec.COMPATIBLE_TLS, ConnectionSpec.CLEARTEXT))
+            .build();
+
+    public Service(String serviceName, InitServiceOptions options) {
+        this.serviceName = serviceName;
+        this.headers = Headers.of(options.getHeaders());
+        this.options = options;
+    }
+
+    private HttpUrl buildUrl(String path, String queryString, InitServiceOptions options) {
+        if (options == null && queryString == null) {
```

Comment:
// Set the values that are the same for each if/else branch

```
builder = new HttpUrl.Builder()
               .host(this.serviceName)
               .port(this.options.getPort())
               .addPathSegments(path)


if (querystring != null)
    builder.query(querystring)

```

etc

Reply:
Done